.build/run-tests-enhanced.sh

```diff
@@ -0,0 +1,454 @@
+        if [ -f "${DIST_DIR}/dtest-${dtest_jar_version}.jar" ] ; then
+            echo "Skipping dtest jar build for branch ${branch} as ${DIST_DIR}/dtest-${dtest_jar_version}.jar already exists"
+            continue
+        fi
+        # redefine the build.dir to local build folder, rightmost definition wins with java command line system properties
+        ant realclean -Dbuild.dir=${TMP_DIR}/cassandra-dtest-jars/build
+        ant jar dtest-jar ${ANT_TEST_OPTS} -Dbuild.dir=${TMP_DIR}/cassandra-dtest-jars/build
+        cp "${TMP_DIR}/cassandra-dtest-jars/build/dtest-${dtest_jar_version}.jar" ${DIST_DIR}/
+    done
+    popd >/dev/null
+    ls -l ${DIST_DIR}/dtest*.jar
+    unset CASSANDRA_USE_JDK11
+}
+
+_run_testlist() {
+    local _target_prefix=$1
+    local _testlist_target=$2
+    local _test_name_regexp=$3
+    local _split_chunk=$4
+    local _test_timeout=$5
+    local _test_iterations=${6:-1}
+
+    # Here we have to deal with several cases
+    #   1. Run repeats: cvs tests
+    #   2. Run split
+    #   3. Run single test
+    if [ "${_test_iterations}" -gt 1 ] ; then
+      # TODO: Support repeat at test method level
+      echo "Running reapted tests: ${_test_name_regexp} chunk ${_split_chunk} ${_test_iterations} times."
+
+      #Repeat tests can come in csv
+      for i in ${_test_name_regexp//,/ }; do
+          testlist="${testlist}"$'\n'"$( _list_tests "${_target_prefix}" | _split_tests "${i}")"
+      done
+    elif [ -n "${_test_name_regexp}" ]; then
+      echo "Running test: ${_test_name_regexp}"
+      testlist="$( _list_tests "${_target_prefix}" | _split_tests "${_test_name_regexp}")"
+    elif [ -n "${_split_chunk}" ]; then
+      echo "Running split: ${_split_chunk}"
+      testlist="$( _list_tests "${_target_prefix}" | _split_tests "${_split_chunk}")"
+    fi
+
+    if [[ "${_split_chunk}" =~ ^[0-9]+/[0-9]+$ ]]; then
+      if [[ -z "${testlist}" ]]; then
+        # something has to run in the split to generate a junit xml result
+        echo "Hacking ${_target_prefix} ${_testlist_target} to run only first test found as no tests in split ${_split_chunk} were found"
+        testlist="$( _list_tests "${_target_prefix}" | head -n1)"
+      fi
+    else
+      if [[ -z "${testlist}" ]]; then
+        echo "No tests match ${_test_name_regexp} ${_split_chunk}"
+        die
+      fi
+    fi
+
+    local -r _results_uuid=$(cat /proc/sys/kernel/random/uuid)
+    for ((i=0; i < _test_iterations; i++)); do
+      if ant "$_testlist_target" -Dtest.classlistprefix="${_target_prefix}" -Dtest.classlistfile=<(echo "${testlist}") -Dtest.timeout="${_test_timeout}" "${ANT_TEST_OPTS}"; then
+
+        if [ "${_test_iterations}" -gt 1 ]; then
+          # Get this test results and rename file with iteration and 'pass'
+          find "${DIST_DIR}"/test/output/ -type f -not -name "*pass.xml" -print0 | while read -r -d $'\0' file; do
+              mv "${file}" "${file%.xml}-${_results_uuid}-${i}-pass.xml"
```

Comment:
won't we need to sed the file too (for when it gets aggregated into other files)

Reply:
I did try `-Dtesttag.extra=` but discarded it. IIRC all test logs, xmls, etc iterations being equally named would overwrite or you'd get N folders instead of N files in one folder. Something along those lines iirc. I can retry if you need the details.